docs/api/csf/csf-factories.mdx

```diff
@@ -0,0 +1,305 @@
+
+{/* prettier-ignore-start */}
+<CodeSnippets path="csf-factories-preview-example.md" />
+{/* prettier-ignore-end */}
+
+<Callout variant="info">
+  The preview configuration will be automatically updated to reference the necessary addons when installing an addon via `npx storybook add <addon-name>` or running `storybook dev`.
+</Callout>
+
+### `preview.meta`
+
+The `meta` function on the `preview` object is used to define the metadata for your stories. It accepts an object containing the `component`, `title`, `parameters`, and other story properties. 
+
+{/* TODO: Snippetize */}
+```ts title="Button.stories.ts"
+// Learn about the # subpath import: https://storybook.js.org/docs/api/csf/csf-factories#subpath-imports
+import preview from '#.storybook/preview';
+
+const meta = preview.meta({
+  component: Button,
+  parameters: {
+    // type-safe!
+    layout: 'centered',
+  }
+});
+export default meta;
+```
+
+### `meta.story`
+
+Finally, the `story` function on the `meta` object is used to define the stories themselves. This function accepts an object containing the `name`, `args`, `parameters`, and other story properties.
+
+{/* TODO: Snippetize */}
+```ts title="Button.stories.ts"
+// Learn about the # subpath import: https://storybook.js.org/docs/api/csf/csf-factories#subpath-imports
+import preview from '#.storybook/preview';
+
+const meta = preview.meta({
+  component: Button,
+  // ...
+});
+
+export const Primary = meta.story({
+  args: {
+    // type-safe!
+    primary: true,
+  },
+});
+```
+
+### Subpath imports
+
+CSF Factories leverages subpath imports to simplify importing constructs from the preview file. While you can still use relative path imports, subpath imports offer a more convenient and maintainable approach:
+
+```ts
+// ✅ Subpath imports won't break if you move story files around
+import preview from '#.storybook/preview';
+
+// ❌ Relative imports will break if you move story files around
+import preview from '../../../.storybook/preview';
+```
+
+The [CSF Factories codemod](#codemod) will automatically add the necessary subpath import for the preview file, or you can [add it manually](#1-add-subpath-import-in-packagejson).
+
+For more details, refer to the [subpath imports documentation](../../writing-stories/mocking-data-and-modules/mocking-modules.mdx#subpath-imports).
+
+## Upgrading from CSF 3
+
+You can upgrade your project's story files to CSF Factories incrementally or all at once. However, you must upgrade your `.storybook/main.ts` and `.storybook/preview.ts` files first before you can use CSF Factories in a story file.
+
+<Callout variant="info">
+  If your story files are not using CSF 3 already, please run `npx storybook migrate csf-2-to-3` to automatically upgrade your project from CSF 2 to CSF 3.
+
+  You can also [migrate manually](./index.mdx#upgrading-from-csf-2-to-csf-3).
+</Callout>
```

Comment:
We can now change/remove this section! The codemod will also migrate from CSF 2 to 3.

Reply:
And CSF 1, presumably? Because that's just CSF 2 without `args`?